### PR TITLE
[FLINK-1976][gelly] Added ForwardedFields Annotations

### PR DESCRIPTION
**Pull request by andralungu**

### CODE REVIEW

**Comment by hsaputra** [comment]
@andralungu , please add more description about the PR to help people review and understand why this PR is made.

**Comment by andralungu** [comment]
Sure @hsaputra , 

I thought the JIRA issue was descriptive enough. We were missing ForwardedFields* annotations for the majority of Gelly classes in Graph.java especially. 
This annotation, as I am sure you know, helps the optimizer generate more efficient execution plans. 

Placing it in front of the classes is optional, but I believe it helps. 
@vasia and @fhueske also agreed. In fact this was Fabian's observation when surfing the Gelly code. Thanks again, Fabian! :) 

I hope things are clearer now, @hsaputra. Let me know if you have further questions :)

**Comment by hsaputra** [comment]
Thx @andralungu, really appreciate it.
Usually JIRA issue describes what is the problem or task to do, and pull request description should tell how and why certain changes were made in the patch to help reviewers understand the context and your line of thoughts.

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java [hunk 1]
```diff
@@ -1329,6 +1355,7 @@ public void join(Tuple3<K, K, Edge<K, EV>> keysWithEdge, Vertex<K, VV> neighbor,
 		}
 	}
 
+	@ForwardedFieldsFirst("f0; f2->f1")
```
Comment (fhueske) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java]: Add `@ForwardedFieldsSecond("*->f2")`?

**Comment by fhueske** [comment]
The annotations look good to me. 
I think one more annotation could be added.

**Comment by andralungu** [comment]
Indeed I was missing that annotation. Nice catch!
PR updated :)

**Comment by andralungu** [comment]
Guys, 

Maybe it makes sense to merge this :)
It's been here for a while.

**Comment by fhueske** [comment]
Thanks for pinging. 
I'll merge it.

**Pull request closed without merging**